lib/racecar/consumer_set.rb

```diff
@@ -142,7 +142,12 @@ def current_subscription
     end
 
     def reset_current_consumer
-      @consumers[@consumer_id_iterator.peek] = nil
+      current_consumer_id = @consumer_id_iterator.peek
+      @logger.info "Resetting consumer with id: #{current_consumer_id}"
+
+      consumer = @consumers[current_consumer_id]
+      consumer.close unless consumer.nil?
```

Comment:
Can `#close` raise an exception? Perhaps we should rescue and log any Rdkafka exception raised here rather than crash the process.

Reply:
I see your point about catching an exception because `#reset_current_consumer` is called after exceptions so that raises the chances of unpredictable behavior. However, [#close](https://github.com/appsignal/rdkafka-ruby/blob/532afaa12aab6341bf4410324dee90055d9db1b6/lib/rdkafka/consumer.rb#L22-L29) does not raise an exception as far as I can tell:
* The docs in [Rdkafka::Consumer#close](https://github.com/appsignal/rdkafka-ruby/blob/532afaa12aab6341bf4410324dee90055d9db1b6/lib/rdkafka/consumer.rb#L22-L29) do not indicate that this method raises anything. The docs for other methods like [poll](https://github.com/appsignal/rdkafka-ruby/blob/532afaa12aab6341bf4410324dee90055d9db1b6/lib/rdkafka/consumer.rb#L414-L420) do show that they raises can error.
* `Rdkafka::Consumer#close` calls [rd_kafka_consumer_close](https://github.com/edenhill/librdkafka/blob/dbafbb748c116b03a18b42a10b845eeb5517d03e/src/rdkafka.h#L3660-L3675) and that always returns an integer even on fatal errors
* `Rdkafka::Consumer#close` also calls calls [rd_kafka_destroy](https://github.com/edenhill/librdkafka/blob/dbafbb748c116b03a18b42a10b845eeb5517d03e/src/rdkafka.h#L2699-L2714) which could `trigger consumer callbacks, such as rebalance_cb` which makes this less predictable.

Given that `#close` does raises `Rdkafka::RdkafkaError` we would have to rescue `StandardError` instead. Does that make sense? 

P.s., I found [these librdkafka docs ](https://github.com/edenhill/librdkafka/blob/acca6ed2f96b4a49d13a801e0dae8d373dd847a1/INTRODUCTION.md#high-level-kafkaconsumer) which describes why not not closing the consumer caused the memory leak.

Reply:
Can we rescue `Rdkafka::RdkafkaError` or were you saying that `#close` does _not_ raise that?

Reply:
`#close` does not raise `Rdkafka::RdkafkaError` so I don't think we need to catch it.

Reply:
:+1: